crates/pipeline_manager/src/api.rs

```diff
@@ -723,6 +723,9 @@ struct NewProgramRequest {
     /// SQL code of the program.
     #[schema(example = "CREATE TABLE Example(name varchar);")]
     code: String,
+    /// Compile the program in JIT mode.
+    #[serde(default)]
+    jit_mode: bool,
```

Comment:
As an API, similar to `PipelineConfig`, this is the right point in time to introduce a `ProgramConfig` which in the DB schema is an opaque YAML value (but is the sharper `ProgramConfig` type everywhere outside the DB schema). It can for now have an enum field called mode (which can be `Static` or `Jit`).

`ProgramConfig` will allow us to evolve program configuration options without needing new migrations or patching the API broadly each time.

Reply:
I am not convinced this is the right approach.  The main reason we model pipeline config at YAML is because it is mostly interpreted by the pipeline itself, not the pipeline manager.  This is not the case for programs, including the new flag. Making sure that the DB schema reflects the semantics of the application is generally a good idea.

Reply:
There's three separate issues here. First is what the REST API looks like. Second is the schema in the DB. Third is who the downstream consumers of these fields are.

* I'd like all REST API objects to be consistent. Pipelines and Connectors currently have an associated `config`. It'd be consistent to have a `config` for Programs too rather than inlining in config options. Having such a config also makes it so we don't need to patch the whole call chain each time we need to add a new configuration parameter.

* Storage in the schema. I don't feel too strongly about this being a sharper type than Yaml. Keeping it yaml does make it so that we will need fewer DB migrations and adding nested fields will be easier going forward, but I'm fine either way.

* It's not the case that the configuration options are mainly interpreted by the pipeline. `PipelineConfig` is interpreted by the cloud runner to configure resource requirements for pods. Some of the connector configurations will also be interpreted to configure things like secrets as well as external services. This `ProgramConfig` will also interpreted by the runners as you already have in this patch (with the runner checking for jit mode).